### PR TITLE
Fix Eth URL client side + migration to remove duplicate chain node

### PR DESCRIPTION
**Pull request by timolegros**
## Link to Issue
Closes: #4283 
Closes: #4253 

## Description of Changes
- Updates the client-side URL in the create community form to the correct https url instead of WSS.
- Adds a migration which removes the duplicate ChainNode and redirects the chains that pointed to it to the correct ChainNode.

## Test Plan
- Test creating an ERC20 community -> verify in the database that the new `chain` has `chain_node_id = 37`
- Test creating an Ethereum starter community -> verify in the database that the new `chain` has `chain_node_id = 37`
- Run `SELECT * FROM "ChainNodes" WHERE eth_chain_id = 1 AND id != 37;` against the commonwealth database and ensure that it returns no results.
- Run 
  ```
  SELECT chain_node_id
  FROM "Chains"
  WHERE created_at IS NOT NULL
  ORDER BY created_at DESC
  LIMIT 2;
  ``` 
  and ensure both `chain_node_id` are 37.

## Deployment Plan

1. Normal deployment to CW

## Other Considerations
The bug that is fixed in this PR prevents the `load-db.sh` script from properly updating the ETH Alchemy URLs locally thus testing this PR is not possible locally. To properly run through the Test Plan it needs to be on Frack or Frick. Once this PR is merged + deployed everyone will need to get a new dump before the `load-db.sh` script will properly update local Alchemy URLs thus enabling all local testing that uses Alchemy URLs.

### CODE REVIEW

**Comment by timolegros** [comment]
In response to your [comment][1] @MuonShot and your [comment][2] @ianrowan (responding here since it is more relevant to this PR) I don't believe that a fix for guardrails should be added to this PR since this is a simple/clean fix that can be merged/deployed quickly to resolve the issue (but I can add the changes if you think it's best @MuonShot)

To place guardrails/checks here is what we would do:
1. Replace the hardcoded urls with hardcoded ChainNode names (which are guaranteed to never change -> e.g. Ethereum or Near will never change names).
2. In the `createChain` route fetch the `ChainNode` with the associated name and link it to the new chain.

@ianrowan While your proposed solution of using hardcoded ChainNode ids would work I think using hardcoded names is better since it means we could totally delete and recreate the ChainNode (meaning it has a different id but the same name) and it would still work whereas with hardcoded ids it wouldn't. Both methods achieve the same result but using the name is more concrete. This also aligns with the incoming changes in the semantics refactor which centers chain event origins around static ChainNode names rather than ChainNode ids or community ids.


[1]: https://github.com/hicommonwealth/commonwealth/pull/3999#issuecomment-1595283265
[2]: https://github.com/hicommonwealth/commonwealth/pull/3999#issuecomment-1595328721

**Comment by CowMuon** [comment]
Let's discuss given our stated preference to ship migrations in separate deployments from our regular cadence.

File: packages/commonwealth/server/migrations/20230616214402-remove-duplicate-eth-chain-node.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    await queryInterface.sequelize.transaction(async (t) => {
+      await queryInterface.bulkUpdate(
```
Comment (jnaviask) [inline packages/commonwealth/server/migrations/20230616214402-remove-duplicate-eth-chain-node.js:6]: Operations need to be transactionalized.

File: packages/commonwealth/server/migrations/20230616214402-remove-duplicate-eth-chain-node.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        'Chains',
+        {
+          chain_node_id: 37,
+        },
+        {
+          chain_node_id: 1285,
```
Comment (jnaviask) [inline packages/commonwealth/server/migrations/20230616214402-remove-duplicate-eth-chain-node.js:12]: What's the reason for this odd chain_node_id?

File: packages/commonwealth/server/migrations/20230616214402-remove-duplicate-eth-chain-node.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        }
+      );
+
+      await queryInterface.bulkDelete('ChainNodes', {
+        id: 1285,
```
Comment (jnaviask) [inline packages/commonwealth/server/migrations/20230616214402-remove-duplicate-eth-chain-node.js:17]: What ChainNode does this id refer to?

**Review by jnaviask — DISMISSED** [review]

**Comment by CowMuon** [comment]
Will start off by saying would like to get this in today. Having said that, would (obvi) like to see mo beta test coverage, but won't be a blocker for getting this, but could be a fast  follow. 

My preference is to not have these hard coded urls inline to begin with (but rather top level declarations or stored as configuration) following our based principle of clean separation of configuration and code. (But again, won't block this fix, which does work, on that basis.) 

However, not fully transactionalizing the migration here is a blocker to getting this merged. What's a realistic ETA for turning this around please? @timolegros

**Comment by CowMuon** [comment]
Setting to draft for the migration to be transactionalized @timolegros 

Sorry about that! @jpegmullet

File: packages/commonwealth/server/migrations/20230616214402-remove-duplicate-eth-chain-node.js (hunk 1 above)
Comment (timolegros) [reply packages/commonwealth/server/migrations/20230616214402-remove-duplicate-eth-chain-node.js:6]: Done

File: packages/commonwealth/server/migrations/20230616214402-remove-duplicate-eth-chain-node.js (hunk 3 above)
Comment (timolegros) [reply packages/commonwealth/server/migrations/20230616214402-remove-duplicate-eth-chain-node.js]: 1285 is the duplicate Ethereum chain node that was added after someone created an ERC20 community with the incorrect hard-coded chain node URL (wss alchemy URL).

File: packages/commonwealth/server/migrations/20230616214402-remove-duplicate-eth-chain-node.js (hunk 2 above)
Comment (timolegros) [reply packages/commonwealth/server/migrations/20230616214402-remove-duplicate-eth-chain-node.js:12]: This updates all the chains/communities that point to the incorrect/false Ethereum chain node (1285) to point to the correct (37) chain node. Not sure if that answers this question.

**Comment by timolegros** [comment]
> Will start off by saying would like to get this in today. Having said that, would (obvi) like to see mo beta test coverage, but won't be a blocker for getting this, but could be a fast follow.
> 
> My preference is to not have these hard coded urls inline to begin with (but rather top level declarations or stored as configuration) following our based principle of clean separation of configuration and code. (But again, won't block this fix, which does work, on that basis.)
> 
> However, not fully transactionalizing the migration here is a blocker to getting this merged. What's a realistic ETA for turning this around please? @timolegros

Writing tests for this is quite difficult without running tests against a copy of the production database or setting up very (and maybe overly) specific database constraints.

**Review by CowMuon — APPROVED** [review]
Thanks Timothee!

**Pull request merged**